### PR TITLE
MEN-4118: restore ALLOWED_ORIGIN_HOSTS regex match when checking Origin

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: title

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -30,6 +30,14 @@ sed -i -e "s/[@]INCLUDE_SERVER_BLOCKS[@]/include \/usr\/local\/openresty\/nginx\
 if [ -n "$ALLOWED_HOSTS" ]; then
     sed -i -e "s/[@]ALLOWED_HOSTS[@]/$ALLOWED_HOSTS/" /usr/local/openresty/nginx/conf/ssl.nginx.conf
     sed -i -e "s/[@]ALLOWED_HOSTS[@]/$ALLOWED_HOSTS/" /usr/local/openresty/nginx/conf/non-ssl.nginx.conf
+
+    # generate ORIGIN whitelist
+    if [ "$ALLOWED_HOSTS" != "_" ]; then
+        hosts=$(echo $ALLOWED_HOSTS | sed 's/ \{1,\}/|/g' | sed 's/[.]\{1,\}/\\\\\\./g')
```
Comment (merlin-northern) [inline entrypoint.sh:36]: I am having doubts about these sed expressions. can you provide an example of ALLOWED_HOSTS and resulting hsots how it should look like?

File: entrypoint.sh [hunk 2]
```diff
@@ -30,6 +30,14 @@ sed -i -e "s/[@]INCLUDE_SERVER_BLOCKS[@]/include \/usr\/local\/openresty\/nginx\
...
+    else
+        hosts=".*"
+    fi
+    sed -i -e "s/[@]ALLOWED_ORIGIN_HOSTS[@]/$hosts/" /usr/local/openresty/nginx/conf/common.nginx.conf
```
Comment (merlin-northern) [inline entrypoint.sh:40]: who sets `ALLOWED_ORIGIN_HOSTS` to be '_'?
I need some examples, I am a bit out of context.

Comment (tranchitella) [reply entrypoint.sh:40]: It's explained here: https://tracker.mender.io/browse/MEN-4118

File: entrypoint.sh (hunk 1 above)
Comment (tranchitella) [reply entrypoint.sh:36]: `my\.custom\.domain|another\.domain\.com`

**Review by alfrunes — APPROVED** [review]

**Pull request merged**